src/group_theory/group_action/defs.lean

```diff
@@ -207,6 +207,11 @@ lemma smul_mul_smul [monoid α] (r s : M) (x y : α) [is_scalar_tower M α α] [
   (r • x) * (s • y) = (r * s) • (x * y) :=
 by rw [smul_mul_assoc, mul_smul_comm, smul_smul]
 
+lemma smul_smul_smul_comm {X Y : Type*} [has_scalar X X] [has_scalar X Y] [has_scalar Y Y]
```

Comment:
I'll do some experiments with:
```lean
@[to_additive] -- Maybe not `to_additive` actually
instance has_mul.to_has_scalar (α : Type*) [has_mul α] : has_scalar α α := ⟨(*)⟩
```
now but I worry about the impact something like this could have.

Reply:
I don't expect it to cause any issues, and should unify the two different `smul_eq_mul` lemmas we have for `monoid.to_mul_action` and `mul_zero_class.to_smul_with_zero`

Reply:
OK I've just added https://github.com/leanprover-community/mathlib/pull/7885/commits/8b9bfe25d1d814f9229044ca40e48f7e877c2168 as a first step in this direction. I expect the build to fail (because I've also generalised `mul_action`) but if I'm lucky it won't be too bad.